.github/workflows/kind.yml

```diff
@@ -0,0 +1,149 @@
+      with:
+        repository: stolostron/governance-policy-framework
+        path: framework
+        ref: ${{ github.event.pull_request.base.ref }} # like main or release-*
+
+    - name: Checkout Policy Framework
+      # Checkout like "usual" when called from this repository
+      if: ${{ github.event.repository.name == 'governance-policy-framework' }}
+      uses: actions/checkout@v2
+      with:
+        # `repository` is inferred as the "caller" repository 
+        path: framework
+        # `ref` is inferred as the new commit
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version-file: framework/go.mod
+
+    - name: Bootstrap the KinD Cluster
+      working-directory: framework
+      env:
+        deployOnHub: ${{ matrix.deployOnHub }}
+        KIND_VERSION: ${{ matrix.kind }}
+      run: |
+        echo "::group::make e2e-dependencies"
+        make e2e-dependencies
+        echo "::endgroup::"
+
+        echo "::group::make kind-bootstrap-cluster"
+        make kind-bootstrap-cluster
+        echo "::endgroup::"
+
+        echo "::group::kubectl get pods -A"
+        kubectl get pods -A
+        echo "::endgroup::"
+
+        echo "Saving kubeconfig paths for use in other steps"
+        echo "MANAGED_KUBECONFIG=$(pwd)/kubeconfig_managed" >> $GITHUB_ENV
+        echo "HUB_KUBECONFIG=$(pwd)/kubeconfig_hub" >> $GITHUB_ENV
+        echo "HUB_INTERNAL_KUBECONFIG=$(pwd)/kubeconfig_hub_internal" >> $GITHUB_ENV
+
+    - name: Patch Component Image
+      if: ${{ github.event.repository.name != 'governance-policy-framework' }}
+      working-directory: component
+      env:
+        deployOnHub: ${{ matrix.deployOnHub }}
+        HUB_ONLY: ${{ inputs.hub_only_component }}
+        WATCH_NAMESPACE: managed
+      run: |
+        export MANAGED_CONFIG=${MANAGED_KUBECONFIG}
+        export HUB_CONFIG=${HUB_KUBECONFIG}
+        export HUB_CONFIG_INTERNAL=${HUB_INTERNAL_KUBECONFIG}
+
+        if [[ "${deployOnHub}" == "true" ]] || [[ "${HUB_ONLY}" == "true" ]]; then
+          echo "Using KIND_NAME=hub and the hub config as the default kubeconfig"
+          export KIND_NAME=hub
+          export KUBECONFIG=${HUB_KUBECONFIG}
+        else
+          echo "Using KIND_NAME=managed and the managed config as the default kubeconfig"
+          export KIND_NAME=managed
+          export KUBECONFIG=${MANAGED_KUBECONFIG}
+        fi
+
+        echo "::group::make build-images"
+        make build-images
+        echo "::endgroup::"
+
+        echo "::group::make kind-deploy-controller-dev"
+        make kind-deploy-controller-dev
+        echo "::endgroup::"
+
+    - name: Run e2e tests
+      working-directory: framework
+      env:
+        TEST_ARGS: --json-report=report.json --junit-report=report.xml --output-dir=test-output
+      run: |
+        make e2e-test
+
+    - name: Debug
+      if: ${{ failure() }}
+      working-directory: framework
+      run: |
+        make e2e-debug-kind
```

Comment:
We should make sure the debug commands are ignored in the `e2e-debug-kind` target. (This probably isn't done in `main` yet either.)